analysis/wikitrust_algorithms/text_trust/version.py

```diff
@@ -0,0 +1,251 @@
+"""
+Module containing Version class. Part of the text_trust package.
+
+Eric Vin, 2019-2020
+"""
+
+#Standard library imports
+import math
+from typing import List, Tuple
+
+#text_trust package imports
+from .block import Block
+from .word import Word
+
+#text_diff package imports
+from wikitrust_algorithms.text_diff.edit import Edit
+
+class Version:
+    """
+    Class representing one version of an article at a certain point in time.
+    """
+
+    def __init__(self, word_list: List[Word], edit_list: List[Edit], author_rep: float, \
+                 constants: Tuple[float, float, float]) \
+                 -> None:
+        """
+        Version Constructor:
+            -words: A list of Words present in this Version
+            -edit_list: A list of Edits that map the previous Version
+                to this Version.
+            -author_rep: The reputation of the author who made the
```

Comment:
The author reputation depends on many revisions. Why write it in a revision?  Also, an author does many revisions; do we want to update this reputation in all revisions (even all revisions in memory) each time the author reputation changes?

Reply:
When writing this I was planning on calculating all edits in order by timestamp, so the author_rep was the author reputation at the time the revision was made. We will definitely have to change this to implement the arbitrary ordered computations.